src/js/vertical-slide.js

```diff
@@ -6,15 +6,28 @@ const shortCodeSlug = sliderContext.getAttribute('data-shortcode-slug');
 
 // Fetching the data with id
 const fetchSliderData = async (id, slug) => {
-  const response = await fetch(`http://localhost/wp-local/wp-json/wp/v2/vertical_slider/${id}?_fields=id,acf`);
-  const data = await response.json();
-  return data;
+  if (id) {
+    const response = await fetch(`http://localhost/wp-local/wp-json/wp/v2/vertical_slider/${id}?_fields=custom_rest_api`);
+    const data = await response.json();
+    return data;
+  } else if (!id && slug) {
+    const response = await fetch(`http://localhost/wp-local/wp-json/wp/v2/vertical_slider?slug=${slug}&_fields=custom_rest_api`);
+    const data = await response.json();
+    return data[0];
```

Comment:
What if this returns 0 results? Is there any error handling we need to add here? Are we going to end up getting a null reference exception here?